roles/sssd_ad/tasks/main.yml

```diff
@@ -0,0 +1,124 @@
+---
+- name: Sssd_ad | change hostname to match AD
+  ansible.builtin.command: hostnamectl set-hostname {{ host_ad_name | default(omit) }}
+  changed_when: false
+  when:
+    - running_on_server
+
+- name: Sssd_ad | allow password authentication
+  ansible.builtin.lineinfile:
+    path: /etc/ssh/sshd_config
+    state: present
+    regexp: '^PasswordAuthentication no'
+    line: 'PasswordAuthentication yes'
+
+- name: Sssd_ad | allow users authentication
+  ansible.builtin.lineinfile:
+    path: /etc/ssh/sshd_config
+    state: present
+    regexp: '^AllowUsers pulsys'
+    line: '# AllowUsers pulsys'
+  when: running_on_server
+  notify: restart sshd
+
+- name: Sssd_ad | install necessary packages
+  ansible.builtin.apt:
+    name: "{{ item }}"
+    state: present
+    update_cache: true
+  loop:
+    - adcli
+    - krb5-user
+    - libnss-sss
+    - libpam-sss
+    - sssd
+    - sssd-tools
+
+- name: Sssd_ad | configure Kerberos
+  ansible.builtin.blockinfile:
+    path: /etc/krb5.conf
+    mode: "0644"
+    create: true
+    block: |
+      [logging]
+        default = FILE:/var/log/krb5libs.log
+        kdc = FILE:/var/log/krb5kdc.log
+        admin_server = FILE:/var/log/kadmind.log
+
+      [libdefaults]
+        dns_lookup_realm = false
+        dns_lookup_kdc = true
+        ticket_lifetime = 24h
+        renew_lifetime = 7d
+        forwardable = true
+        rdns = false
+        default_realm = {{ ad_domain | upper }}
+        default_ccache_name = KEYRING:persistent:%{uid}
+
+      [realms]
+        {{ ad_domain | upper }} = {
+          kdc = {{ ad_domain_controller }}
+          admin_server = {{ ad_domain_controller }}
+        }
+
+      [domain_realm]
+        .{{ ad_domain | lower }} = {{ ad_domain | upper }}
+        {{ ad_domain | lower }} = {{ ad_domain | upper }}
+
+- name: Sssd_ad | create sssd.conf
+  ansible.builtin.template:
+    src: sssd.conf.j2
+    dest: /etc/sssd/sssd.conf
+    mode: "0644"
+
+- name: Sssd_ad | enable and start sssd service
+  ansible.builtin.service:
+    name: sssd
+    state: started
+    enabled: true
+  register: sssd_service
+  when: running_on_server
+
+- name: Sssd_ad | join the domain
+  ansible.builtin.command: realm join {{ ad_domain }} -U {{ ad_admin_user }} --install=/
+  environment:
+    KRB5_CONFIG: /etc/krb5.conf
+  changed_when: false
+  when: running_on_server
+
+- name: Sssd_ad | verify domain join
+  ansible.builtin.command: realm list
+  register: realm_list
+  changed_when: false
+  when: running_on_server
+
+- name: Sssd_ad | debug realm list output
```

Comment:
Let's change this so it stops the playbook if the output is not correct at this stage.

Reply:
only way to know what this output is will make this PR larger by incorporating this. https://github.com/pulibrary/princeton_ansible/tree/i4938_sftp_role